libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/instruction/InstructionModel.java

```diff
@@ -63,34 +88,45 @@ RouteProgress getProgress() {
 
   private void buildInstructionModel(RouteProgress progress, DecimalFormat decimalFormat) {
     formatStepDistance(progress, decimalFormat);
-    LegStep upComingStep = progress.currentLegProgress().upComingStep();
-    if (upComingStep != null) {
-      extractStepResources(upComingStep);
+    LegStep upcomingStep = progress.currentLegProgress().upComingStep();
+    if (upcomingStep != null) {
+      extractStepResources(progress, upcomingStep);
     }
   }
 
-  private void extractStepResources(LegStep upComingStep) {
-    if (hasManeuver(upComingStep)) {
-      maneuverModifier = upComingStep.maneuver().modifier();
-      maneuverType = upComingStep.maneuver().type();
+  private void extractStepResources(RouteProgress progress, LegStep upcomingStep) {
+    // Upcoming step
+    if (hasManeuver(upcomingStep)) {
+      upcomingStepManeuverType = upcomingStep.maneuver().type();
+      upcomingStepManeuverModifier = upcomingStep.maneuver().modifier();
     }
-    if (hasIntersections(upComingStep)) {
-      intersectionTurnLanes(upComingStep);
+
+    // Then step (step after upcoming)
+    LegStep thenStep = extractThenStep(progress, upcomingStep);
+    if (thenStep != null && hasManeuver(thenStep)) {
+      thenStep(upcomingStep, thenStep);
+    }
+
+    // Turn lane data
+    if (hasIntersections(upcomingStep)) {
+      intersectionTurnLanes(upcomingStep);
```

Comment:
We'll want to update this later to only show at the approach announcement. When we take this second pass, turn lanes, when available, should override the then step.